src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,103 @@
+package duke.parser;
+
+import java.util.List;
+import duke.task.*;
+import duke.command.*;
+import duke.exception.*;
+import duke.ui.*;
+
+public class Parser {
+    public static Command<List<Task>> parseInput(String userInput) throws DukeException {
+        try {
+            String[] parts = userInput.split(" ", 2);
+            String command = parts[0].toLowerCase();
+            List<String> arguments = parts.length > 1 ? List.of(parts[1].split("\\s+")) : List.of();
+
+            String taskDescription = extractTaskDescription(arguments);
+            String deadline = "Error: No deadline set.";
+            String start = "Error: No start time set.";
+            String end = "Error: No end time set.";
+
+            // Get Deadline
```

Comment:
Please give more detailed comments for better clarity as to what you are trying to do in this instance!